state/metrics.go

```diff
@@ -2,41 +2,402 @@ package state
+	pfbSubmissionObservable, err := meter.Int64ObservableCounter(
+		"state_pfb_submission_total_observable",
+		metric.WithDescription("Observable total number of PayForBlob submissions"),
+	)
+	if err != nil {
+		return err
+	}
+
+	gasEstimationObservable, err := meter.Int64ObservableCounter(
+		"state_gas_estimation_total_observable",
+		metric.WithDescription("Observable total number of gas estimation operations"),
+	)
+	if err != nil {
+		return err
+	}
+
+	gasPriceEstimationObservable, err := meter.Int64ObservableCounter(
+		"state_gas_price_estimation_total_observable",
+		metric.WithDescription("Observable total number of gas price estimation operations"),
+	)
+	if err != nil {
+		return err
+	}
+
+	accountQueryObservable, err := meter.Int64ObservableCounter(
+		"state_account_query_total_observable",
+		metric.WithDescription("Observable total number of account query operations"),
+	)
+	if err != nil {
+		return err
+	}
 
 	callback := func(_ context.Context, observer metric.Observer) error {
-		observer.ObserveInt64(pfbCounter, ca.PayForBlobCount())
-		observer.ObserveInt64(lastPfbTimestamp, ca.LastPayForBlob())
+		// Observable metrics for OTLP export
+		observer.ObserveInt64(pfbSubmissionObservable, m.totalPfbSubmissions.Load())
+		observer.ObserveInt64(gasEstimationObservable, m.totalGasEstimations.Load())
+		observer.ObserveInt64(gasPriceEstimationObservable, m.totalGasPriceEstimations.Load())
+		observer.ObserveInt64(accountQueryObservable, m.totalAccountQueries.Load())
 		return nil
 	}
 
-	clientReg, err := meter.RegisterCallback(callback, pfbCounter, lastPfbTimestamp)
+	clientReg, err := meter.RegisterCallback(
+		callback,
+		pfbSubmissionObservable,
+		gasEstimationObservable,
+		gasPriceEstimationObservable,
+		accountQueryObservable,
+	)
+	if err != nil {
+		log.Errorf("Failed to register metrics callback: %v", err)
+		return err
+	}
+
+	// Update the CoreAccessor with the new metrics
+	ca.metrics = m
+
+	// Store the client registration for cleanup
+	m.clientReg = clientReg
+
+	return nil
+}
+
+// Stop cleans up the metrics resources
+func (m *metrics) Stop() error {
+	if m == nil || m.clientReg == nil {
+		return nil
+	}
+	if err := m.clientReg.Unregister(); err != nil {
+		log.Warnw("failed to close metrics", "err", err)
+		return err
+	}
+	return nil
+}
+
+// ObservePfbSubmission records PayForBlob submission metrics
+func (m *metrics) ObservePfbSubmission(
+	ctx context.Context,
+	duration time.Duration,
+	blobCount int,
+	totalSize int64,
+	gasEstimationDuration time.Duration,
+	gasPrice float64,
+	err error,
+) {
+	if m == nil {
+		return
+	}
+
+	// Update counters
+	m.totalPfbSubmissions.Add(1)
+	if err != nil {
+		m.totalPfbSubmissionErrors.Add(1)
+	}
+
+	// Record metrics
+	attrs := []attribute.KeyValue{
+		attribute.Int("blob_count", blobCount),
+		attribute.Int64("total_size_bytes", totalSize),
+		attribute.Float64("gas_price_utia", gasPrice),
+		attribute.Bool("success", err == nil),
```

Comment:
Ideally we would define some error types here too, but that can be done as follow up once we enumarate all of them